app/main.py

```diff
@@ -1,2 +1,57 @@
+from typing import Any, Hashable
+
+
+class Node:
+    def __init__(self, key: Hashable, value: Any) -> None:
+        self.key = key
+        self.value = value
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.capacity = 8
+        self.hash_table: list = [None] * 8
+        self.load_capacity = 2 / 3
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self.get_index(key)
+        while self.hash_table[index]:
+            if self.hash_table[index].key == key:
+                self.hash_table[index].value = value
+                return
+            index = (index + 1) % self.capacity
+        if self.length + 1 > self.capacity * self.load_capacity:
+            self.resize()
+            return self.__setitem__(key, value)
+        self.hash_table[index] = Node(key, value)
+        self.length += 1
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = self.get_index(key)
+        while (self.hash_table[index]
+               and self.hash_table[index].key != key):
+            index = (index + 1) % self.capacity
+        if not self.hash_table[index]:
+            raise KeyError(f"Key {key} is not found")
+        return self.hash_table[index].value
+
+    def __len__(self) -> int:
+        return self.length
+
+    def resize(self) -> None:
+        old_nodes = []
+        for node in self.hash_table:
+            if node:
+                old_nodes.append(node)
+        self.capacity = self.capacity * 2
+        new_hash_table = [None] * self.capacity
+        for node in old_nodes:
+            index = self.get_index(node.key)
```

Comment:
Same as 18, 31. Try to optimize this.
Also try to optimize resize function, it can be simplified